### PR TITLE
Enhancing model-mapper: adding bi-directional caches

### PR DESCRIPTION
**Pull request by yue9944882**

### CODE REVIEW

File: util/src/main/java/io/kubernetes/client/util/ModelMapper.java [hunk 1]
```diff
@@ -114,8 +115,34 @@ public static void addModelMap(String group, String version, String kind, Class<
    */
   public static void addModelMap(
       String group, String version, String kind, String resourceNamePlural, Class<?> clazz) {
+    // TODO(yue9944882): consistency between bi-directional maps
```
Comment (brendandburns) [inline util/src/main/java/io/kubernetes/client/util/ModelMapper.java:118]: I would just add a wrapper class that contains two `ConcurrentHashMap` and implements `Map` wdyt?

File: util/src/main/java/io/kubernetes/client/util/ModelMapper.java [hunk 2]
```diff
@@ -53,12 +53,13 @@
   // Model's api-version midfix to kubernetes api-version
   private static List<String> preBuiltApiVersions = new ArrayList<>();
 
-  // TODO(yue9944882): make the map bi-directional
   private static Map<GroupVersionKind, Class<?>> classesByGVK = new ConcurrentHashMap<>();
-
-  // TODO(yue9944882): make the map bi-directional
   private static Map<GroupVersionResource, Class<?>> classesByGVR = new ConcurrentHashMap<>();
 
+  private static Map<Class<?>, GroupVersionKind> gvkByClasses = new ConcurrentHashMap<>();
```
Comment (brendandburns) [inline util/src/main/java/io/kubernetes/client/util/ModelMapper.java:59]: I think we should encapsulate these two maps into a single bi-directional map. Ok for this to be a TODO.

**Comment by brendandburns** [comment]
Small comments, also tests are failing.

File: util/src/main/java/io/kubernetes/client/util/ModelMapper.java [hunk 3]
```diff
@@ -325,4 +362,22 @@ private static void initModelMap() throws IOException {
         .findFirst()
         .orElse(new MutablePair(null, name));
   }
+
+  static class BiDirectionalMap<K, V> {
+    private Map<K, V> kvMap = new HashMap<>();
```
Comment (brendandburns) [inline util/src/main/java/io/kubernetes/client/util/ModelMapper.java:367]: I think we want this to be `ConcurrentHashMap<>()` ?

Comment (yue9944882) [reply util/src/main/java/io/kubernetes/client/util/ModelMapper.java:367]: the getter/setter of the bi-directional-map are locked by `synchronized` (also to keep the consistency between the two maps) so a plain hash-map is sufficient, isn't it?

**Pull request merged**